core/trino-spi/src/main/java/io/trino/spi/connector/Constraint.java

```diff
@@ -41,17 +44,36 @@ public static Constraint alwaysFalse()
 
     public Constraint(TupleDomain<ColumnHandle> summary)
     {
-        this(summary, Optional.empty(), Optional.empty());
+        this(summary, TRUE, Optional.empty(), Optional.empty());
     }
 
     public Constraint(TupleDomain<ColumnHandle> summary, Predicate<Map<ColumnHandle, NullableValue>> predicate, Set<ColumnHandle> predicateColumns)
     {
-        this(summary, Optional.of(predicate), Optional.of(predicateColumns));
+        this(summary, TRUE, Optional.of(predicate), Optional.of(predicateColumns));
     }
 
-    private Constraint(TupleDomain<ColumnHandle> summary, Optional<Predicate<Map<ColumnHandle, NullableValue>>> predicate, Optional<Set<ColumnHandle>> predicateColumns)
+    public Constraint(TupleDomain<ColumnHandle> summary, ConnectorExpression expression)
+    {
+        this(summary, expression, Optional.empty(), Optional.empty());
+    }
+
+    public Constraint(
+            TupleDomain<ColumnHandle> summary,
+            ConnectorExpression expression,
+            Predicate<Map<ColumnHandle, NullableValue>> predicate,
+            Set<ColumnHandle> predicateColumns)
+    {
+        this(summary, expression, Optional.of(predicate), Optional.of(predicateColumns));
+    }
+
+    private Constraint(
+            TupleDomain<ColumnHandle> summary,
+            ConnectorExpression expression,
```

Comment:
In `ConnectorMetadata#applyAggregation` we have explicit `Map<String, ColumnHandle> assignments` for mapping from `ConnectorExpression`'s `Variable` to `ColumnHandle` in the connector.

What's the mapping here?
Is the difference intentional?

Reply:
Added

Reply:
thanks!